src/sentry/search/events/fields.py

```diff
@@ -148,8 +149,17 @@ def project_threshold_config_expression(organization_id, project_ids):
         .values("project_id", "threshold", "metric")
     )
 
-    if not threshold_configs.count():
+    num_configured = threshold_configs.count()
+    sentry_sdk.set_tag("project_threshold.count", num_configured)
+    sentry_sdk.set_tag(
+        "project_threshold.count.grouped",
+        format_grouped_length(num_configured, [10, 100, 250, 500]),
+    )
+
+    if num_configured == 0:
         return ["tuple", [f"'{DEFAULT_PROJECT_THRESHOLD_METRIC}'", DEFAULT_PROJECT_THRESHOLD]]
+    elif num_configured > MAX_QUERYABLE_TRANSACTION_THRESHOLDS:
+        raise InvalidSearchQuery("Too many configured thresholds, try with fewer projects.")
```

Comment:
Can we reword this error message to tell the user what the limit is so they knows how to change the project selection?